src/pyenphase/envoy.py

```diff
@@ -374,3 +388,41 @@ async def close_dry_contact(self, id: str) -> dict[str, Any]:
             assert self.data is not None  # nosec
         self.data.dry_contact_status[id].status = DryContactStatus.CLOSED
         return result
+
+    async def enable_charge_from_grid(self) -> dict[str, Any]:
+        """Enable charge from grid for Encharge batteries."""
+        if not self.supported_features & SupportedFeatures.ENCHARGE:
+            raise EnvoyFeatureNotAvailable(
+                "This feature requires Enphase Encharge or IQ Batteries."
+            )
+        if not self.data or not self.data.tariff:
+            raise ValueError(
+                "Tried to enable charge from grid before the Envoy was queried."
+            )
+        if not self.data.tariff.storage_settings:
+            raise EnvoyFeatureNotAvailable(
+                "This feature requires Enphase Encharge or IQ Batteries."
+            )
```

Comment:
This could be a function `self._verify_encharge_storage_or_raise`

Reply:
This causes mypy to complain:
src/pyenphase/envoy.py:390: error: Item "None" of "Optional[EnvoyData]" has no attribute "tariff"  [union-attr]
src/pyenphase/envoy.py:390: error: Item "None" of "Union[EnvoyTariff, None, Any]" has no attribute "storage_settings"  [union-attr]
src/pyenphase/envoy.py:390: error: Item "None" of "Union[EnvoyStorageSettings, None, Any]" has no attribute "charge_from_grid"  [union-attr]
src/pyenphase/envoy.py:392: error: Item "None" of "Optional[EnvoyData]" has no attribute "tariff"  [union-attr]
src/pyenphase/envoy.py:392: error: Item "None" of "Union[EnvoyTariff, None, Any]" has no attribute "to_api"  [union-attr]
src/pyenphase/envoy.py:398: error: Item "None" of "Optional[EnvoyData]" has no attribute "tariff"  [union-attr]
src/pyenphase/envoy.py:398: error: Item "None" of "Union[EnvoyTariff, None, Any]" has no attribute "storage_settings"  [union-attr]
src/pyenphase/envoy.py:398: error: Item "None" of "Union[EnvoyStorageSettings, None, Any]" has no attribute "charge_from_grid"  [union-attr]
src/pyenphase/envoy.py:400: error: Item "None" of "Optional[EnvoyData]" has no attribute "tariff"  [union-attr]
src/pyenphase/envoy.py:400: error: Item "None" of "Union[EnvoyTariff, None, Any]" has no attribute "to_api"  [union-attr]